api/common.js

```diff
@@ -413,6 +415,65 @@ exports.load_github_detail = function(service_name, cb) {
     }).catch(cb)
 }
 
+
+exports.generateQuery = function(str){
+    stopwords.add('');
+    let result=stopwords.cleanText(str).split(' ').map((word) => `W='${word}'`).join(',');
+    return "OR("+result+")"
+}
+
+exports.updateProjectMag = function(project,cb){
+    if(!config.mag) return console.log("no mag config");
+    let headers = {
+        'Ocp-Apim-Subscription-Key': config.mag.subscription_key,
+        'User-Agent': 'brainlife',
+    }
+    const query = exports.generateQuery(project.name+" "+project.desc);
+    let urlMag = "https://api.labs.cognitive.microsoft.com/academic/v1.0/evaluate?expr="+query+"&model=latest&count=10&offset=0&attributes=Id,AA.AfId,AA.AfN,AA.AuId,AA.AuN,AA.DAuN,CC,CN,D,Ti,F.FId,F.FN,Y,VFN,DOI,IA"
+
+    const papers = () => axios.get(urlMag,{headers}).then(res =>{
+        if(res.status !=200){
+            console.error(res.status);
+            return cb("Failed to get related papers");
+        }
+        return res.data.entities.filter(papers => papers.logprob > -15);        
+    }).catch(cb);
+
+    let dataMag= []
+    papers().then(papers => {
+        papers.forEach(function(paper){
+            let object = {};
+            let date = Date.parse(paper.D);      // get Date and 
+            object.publicationDate = new Date(date);        
+            object.citationCount = paper.CC; // add Citation Count
+            object.title = paper.Ti; // add Title
+            object.venue = paper.VFN; // add venue
+            object.authors = paper.AA.map(function(author){
+                const modifiedAuthors = {"institution" : author.AfN, "name" : author.DAuN };
+                return modifiedAuthors;
+            })
+            if(paper.F){
+                object.fields = paper.F.map((name) => name.FN).join(',');
+            }
+            if(paper.IA){
+                object.abstract = Object.keys(paper.IA.InvertedIndex).join(' ');
```

Comment:
I think it gives the sentence broken into words : 

Here is sample abstract generated for one of the projects: 

In comparative high-throughput sequencing assays, a fundamental task is the analysis of count data, such as read counts per gene in RNA-seq, for evidence systematic changes across experimental conditions. Small replicate numbers, discreteness, large dynamic range and presence outliers require suitable statistical approach. We present DESeq2, method differential using shrinkage estimation dispersions fold to improve stability interpretability estimates. This enables more quantitative focused on strength rather than mere expression. The DESeq2 package available at http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html\n .

Reply:
Your code outputs this

```
The National Institute of Mental Health strategic plan for advancing psychiatric neuroscience calls an acceleration discovery and the delineation developmental trajectories risk resilience across lifespan. To attain these objectives, sufficiently powered datasets with broad deep phenotypic characterization, state-of-the-art neuroimaging, genetic samples must be generated made openly available to scientific community. enhanced Nathan Kline Rockland Sample (NKI-RS) is a response this need. NKI-RS ongoing, institutionally-centered endeavor aimed at creating large-scale (N>1000), deeply phenotyped, community-ascertained, lifespan sample (ages 6-85 years old) advanced neuroimaging genetics. These data will publically shared, prospectively (i.e., on weekly basis). Herein, we describe conceptual basis NKI-RS, including study design, sampling considerations, steps synchronize assessment. Additionally, our process sharing community while protecting participant confidentiality, maintaining adequate database, certifying integrity. pilot phase challenges in recruiting, characterizing, imaging, data, discussed also explaining how experience informed final design NKI-RS. It hope that familiarity underpinnings facilitate harmonization future collection efforts nosology.
```

Instead, the correct output should look like this

```
The National Institute of Mental Health strategic plan for advancing psychiatric neuroscience calls for an acceleration of discovery and the delineation of developmental trajectories for risk and resilience across the lifespan. To attain these objectives, sufficiently powered datasets with broad and deep phenotypic characterization, state-of-the-art neuroimaging, and genetic samples must be generated and made openly available to the scientific community. The enhanced Nathan Kline Institute Rockland Sample (NKI-RS) is a response to this need. NKI-RS is an ongoing, institutionally-centered endeavor aimed at creating a large-scale (N>1000), deeply phenotyped, community-ascertained, lifespan sample (ages 6-85 years old) with advanced neuroimaging and genetics. These data will be publically shared, openly and prospectively (i.e., on a weekly basis). Herein, we describe the conceptual basis of the NKI-RS, including study design, sampling considerations, and steps to synchronize phenotypic and neuroimaging assessment. Additionally, we describe our process for sharing the data with the scientific community while protecting participant confidentiality, maintaining an adequate database, and certifying data integrity. The pilot phase of the NKI-RS, including challenges in recruiting, characterizing, imaging, and sharing data, is discussed while also explaining how this experience informed the final design of the enhanced NKI-RS. It is our hope that familiarity with the conceptual underpinnings of the enhanced NKI-RS will facilitate harmonization with future data collection efforts aimed at advancing psychiatric neuroscience and nosology.
```

Hint.. :)

```
            //reconstruct abstract
            const abstract = [];
            for(const word in paper.IA.InvertedIndex) {
                paper.IA.InvertedIndex[word].forEach(idx=>{
                    abstract[idx] = word;
                });
            }
```